### PR TITLE
Global features Angular Api added

### PR DESCRIPTION
**Pull request by mahmut-gundogdu**
Resolves  #[12546](https://github.com/abpframework/abp/issues/12546)

@braim23  please, Could you check the docs,?
@muhammedaltug please,  Could you check the angular, please?

### CODE REVIEW

File: docs/en/UI/Angular/Config-State-Service.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { ConfigStateService } from '@abp/ng.core';
   /* class metadata here */
 })
 class DemoComponent {
-  constructor(private config: ConfigStateService) {}
+  constructor(private config: \) {}
```
Comment (muhammedaltug) [inline docs/en/UI/Angular/Config-State-Service.md]: this change seems unnecessary

File: docs/en/UI/Angular/GlobalFeatures.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component({
+  /* class metadata here */
+})
+class DemoComponent {
+  constructor(private config: ConfigStateService) {}
+}
+
+// Gets all enabled global features.
+const getGlobalFeatures = this.config.getGlobalFeatures ();
```
Comment (muhammedaltug) [inline docs/en/UI/Angular/GlobalFeatures.md]: this line should be 
```typescript
const getGlobalFeatures = this.config.getGlobalFeatures();
```

File: docs/en/UI/Angular/GlobalFeatures.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+true
+
+> this.config.getGlobalFeatureIsEnabled('My.Subscription')
+
+false
+
+// or
+this.config.getGlobalFeatureIsEnabled$('Ecommerce.Subscription').subscribe((isEnabled:boolean) => {
+   // use isEnabled here
+})
```
Comment (muhammedaltug) [inline docs/en/UI/Angular/GlobalFeatures.md]: These lines should be in a method like ng0nInit

File: npm/ng-packs/packages/core/src/lib/services/config-state.service.ts [hunk 4]
```diff
@@ -138,6 +138,31 @@ export class ConfigStateService {
...
+  getGlobalFeatures$() {
+    return this.store.sliceState(state => state.globalFeatures);
+  }
+
+  getGlobalFeatureIsEnabled(key: string) {
+    const globalFeatures = this.store.state.globalFeatures;
+
+    if (!(globalFeatures?.enabledFeatures)) { return false };
+
+    return globalFeatures.enabledFeatures.indexOf(key) != -1;
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/core/src/lib/services/config-state.service.ts]: You can use `Array.some` function

File: npm/ng-packs/packages/core/src/lib/services/config-state.service.ts [hunk 5]
```diff
@@ -138,6 +138,31 @@ export class ConfigStateService {
...
+  }
+
+  getGlobalFeatureIsEnabled$(key: string) {
+    return this.store.sliceState(state => {
+      debugger
+      if (!(state.globalFeatures?.enabledFeatures)) { return false };
+      return state.globalFeatures.enabledFeatures.indexOf(key) != -1 || true;
+    });
+  }
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/core/src/lib/services/config-state.service.ts]: These methods can be simplified with the following code
```typescript
getGlobalFeatureIsEnabled(key: string) {
    return this.isGlobalFeatureEnabled(key, this.store.state.globalFeatures);
 }
 getGlobalFeatureIsEnabled$(key: string) {
    return this.store.sliceState(state => this.isGlobalFeatureEnabled(key, state.globalFeatures));
  }

private isGlobalFeatureEnabled(key: string, globalFeatures: ApplicationGlobalFeatureConfigurationDto) {
    const features = globalFeatures.enabledFeatures || []
    return features.some(f => key === f);
}
```

File: npm/ng-packs/packages/core/src/lib/services/config-state.service.ts [hunk 6]
```diff
@@ -138,6 +138,31 @@ export class ConfigStateService {
...
+    const globalFeatures = this.store.state.globalFeatures;
+
+    if (!(globalFeatures?.enabledFeatures)) { return false };
+
+    return globalFeatures.enabledFeatures.indexOf(key) != -1;
+  }
+
+  getGlobalFeatureIsEnabled$(key: string) {
+    return this.store.sliceState(state => {
+      debugger
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/core/src/lib/services/config-state.service.ts]: forgotten debugger

File: docs/en/UI/Angular/GlobalFeatures.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  /* class metadata here */
+})
+class DemoComponent {
+  constructor(private config: ConfigStateService) {}
+}
+
+// Gets all enabled global features.
+const getGlobalFeatures = this.config.getGlobalFeatures ();
+
+// { enabledFeatures: [ 'Shopping.Payment', 'Ecommerce.Subscription' ] }
```
Comment (muhammedaltug) [inline docs/en/UI/Angular/GlobalFeatures.md]: We should add an explanation for this line eg: `Example Response`

**Review by muhammedaltug — APPROVED** [review]

**Review by braim23 — APPROVED** [review]

**Pull request merged**